crates/ide/src/lib.rs

```diff
@@ -423,6 +423,15 @@ impl Analysis {
         self.with_db(|db| hover::hover(db, position, config))
     }
 
+    /// Returns a short text displaying the type of the expression.
+    pub fn hover_range(
```

Comment:
Rather than adding a second function, let's change the existing one to take a range, and let's use empty range for the case where the input is a single position.